internal/servers/controller/handlers/sessions/session_service.go

```diff
@@ -120,19 +124,25 @@ func (s Service) ListSessions(ctx context.Context, req *pbs.ListSessionsRequest)
 		item.Scope = scopeInfoMap[item.GetScopeId()]
 		res.ScopeId = item.Scope.Id
 		authorizedActions := authResults.FetchActionSetForId(ctx, item.Id, IdActions, auth.WithResource(res))
-		if len(authorizedActions) > 0 {
-			onlySelf := true
-			for _, v := range authorizedActions {
-				if v != action.ReadSelf && v != action.CancelSelf {
-					onlySelf = false
-					break
-				}
-			}
-			if !onlySelf || item.GetUserId() == authResults.UserId {
-				item.AuthorizedActions = authorizedActions.Strings()
-				finalItems = append(finalItems, item)
+		if len(authorizedActions) == 0 {
+			continue
+		}
+		onlySelf := true
+		for _, v := range authorizedActions {
+			if v != action.ReadSelf && v != action.CancelSelf {
+				onlySelf = false
+				break
 			}
 		}
+		if onlySelf && item.GetUserId() != authResults.UserId {
+			continue
+		}
+
+		item.AuthorizedActions = authorizedActions.Strings()
```

Comment:
Can you move this into the `if` below? Might as well not generate the strings if it's filtered.

Reply:
Discussed in Slack; it's here so it can be used in the filter